### DISCUSSION TITLE
custom signin with server side component not working

### DISCUSSION THREAD

**Discussion by nizarlj** [nextauthjs/next-auth#8507] [Help]
### Question 💬

Im trying to add a custom signin Page and can't get it to work, i'm unsure if i'm missing something or if it's a bug, as for the SignIn component itself i've basically copied what was provided on the documentation with slight modifications to use a server-component instead for nextjs 13 (https://next-auth.js.org/configuration/pages#credentials-sign-in).
Using the automatically generated signin page by commenting out the pages in the config everything works fine, but for some reason my custom component won't work.

The form action is submitted with the correct payload(username, password, csrfToken), but for reasons unknown to me the authorize method doesn't get called when using the custom signin page, however when using the automatically generated page the credentials get logged as they should.

### How to reproduce ☕️

config:
```ts
export const options: NextAuthOptions = {
    providers: [
        CredentialsProvider({
            name: "Credentials",
            credentials: {
                username: {
                    label: "Username:",
                    type: "text",
                    placeholder: "username"
                },
                password: {
                    label: "Password:",
                    type: "password",
                    placeholder: "password"
                }
            },
            async authorize(credentials) {
                console.log(credentials)
                const user = { id: "1", username: "test", password: "test" };

                if(credentials?.username === user.username && credentials?.password === user.password) {
                    return user;
                }

                return null;
            }
        })
    ],
    pages: {
        signIn: "/signin"
    },
}
```

SignIn.tsx
```ts
export default async function SignIn() {
  const csrfToken = await getCsrfToken();

  return (
    <form method="post" action="/api/auth/callback/credentials">
      <input name="csrfToken" type="hidden" defaultValue={csrfToken} />
      <label>
        Username
        <input name="username" type="text" />
      </label>
      <label>
        Password
        <input name="password" type="password" />
      </label>
      <button type="submit">Signin</button>
    </form>
    )
}
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by rahul-reveation** [comment]
I lost 2 days trying to make custom sign in work. Never figured out the actual issue until I came across this issue https://github.com/nextauthjs/next-auth/discussions/7256

Basically, getCsrfToken will not work on server side component. It doesn't return the correct token value. I made it work by extracting csrfToken from the cookie as recommended in the linked topic.

```
  const cookieStore = cookies();
  const token = cookieStore.get('next-auth.csrf-token');
  const csrfToken = token?.value.split('|')[0];
```

**Reply by zakwear** [reply]
@rahul-reveation yo that worked. Thank you.

**Reply by ssmithsoftware** [reply]
good find!
